doc/help-search.md

```diff
@@ -24,17 +24,150 @@ The search UI also supports filters:
   - On the left of the search bar, select 'Dart' or 'Flutter' depending on the
     SDK you are developing with.
 
-  - On the right of the search bar, click the platform filter buttons to filter
-    results to packages supporting the platforms you need support for.
-
-  - To use platform filters for alpha platforms (e.g. Flutter Desktop), use
-    the Advanced option on the far right of the search bar.
+  - When an SDK filter is selected, on the right of the search bar, click the
+    platform filter buttons to filter results to packages supporting the platforms
+    you need support for.
 
 ## Ranking
 
-Default listings use a composite of the three scores to sort packages. The
-ranking is based on the composite score, and if applicable, the platform
-specificity and the text match score of the search team.
+Package listings (with or without text search, filters or scopes) build on
+a ranking score computed from multiple scores. If applicable, this score is
+adjusted with the text query match score or the scope specificity score
+of the search query.
+
+### Base composite score
+
+The base composite score consist of the following:
+
+  - 50% from pub score (calculated from `pana`)
+  - 50% from usage metrics from likes and popularity
+
+The usage metrics use a non-linear scoring model, where the package above the N-th
+percentile gets N/100 points (e.g. if a package has more likes than the 90% of the
+packages, it will get 0.90 points for likes).
+
+### Scope specificity score
+
+When a platform filter is specified, we both check whether the predicate
+matches the platforms of the package, and also calculate how closely it
+matches them ([`scope_specificity.dart`](https://github.com/dart-lang/pub-dev/blob/master/app/lib/search/scope_specificity.dart)).
+
+For example, if the platform filter is `platform:web`, we'll have the following scores:
+
+- `<missing or empty platform>`: no match, doesn't show up in the results
+- `['platform:android', 'platform:ios']`: no match, doesn't show up in the results
+- `['platform:web']`: `1.0` (specific platform match)
+- `['platform:android', 'platform:web']`: `0.9` (1 extra platform, close platform match)
+- `['platform:android', 'platform:ios', 'platform:web']`: `0.8` (2+ extra platforms, distant platform match)
+
+### Numerical ordering
+
+Numerical ordering is used on search requests that dp not specify text query, but
+would like to order the results by `updated`, or `created` time, and also by
+`likes`, `popularity`, or `pub points`. 
+
+These work on the raw values, without any weights or transformation.
+
+### Text match ranking
+
+The indexing process prepares the package documents and parses the following fields:
+
+- package name
+- package description (first 500 characters)
+- readme (first 5000 characters)
+
+When a text query is specified, we'll try to match the query against all of these
```

Comment:
the word "try" is a bit odd here (hopefully the search does a bit more than trying?)